packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m

```diff
@@ -459,37 +487,47 @@ - (void)updateAutoMediaPlaybackPolicy:(NSNumber*)policy
   }
 }
 
-- (bool)loadRequest:(NSDictionary<NSString*, id>*)request {
-  if (!request) {
-    return false;
+/**
+ * Parses the method call arguments and converts them to an NSURLRequest object.
+ *
+ * @param arguments the method call arguments.
+ *
+ * @return NSURLRequest object.
+ */
+- (NSURLRequest*)buildNSURLRequest:(NSDictionary<NSString*, id>*)arguments {
+  id requestParameters = arguments[@"request"];
+  if (![requestParameters isKindOfClass:[NSDictionary class]]) {
+    return nil;
   }
 
-  NSString* url = request[@"url"];
-  if ([url isKindOfClass:[NSString class]]) {
-    id headers = request[@"headers"];
-    if ([headers isKindOfClass:[NSDictionary class]]) {
-      return [self loadUrl:url withHeaders:headers];
-    } else {
-      return [self loadUrl:url];
-    }
+  NSString* urlString = requestParameters[@"uri"];
+  if (!urlString) {
+    return nil;
   }
 
-  return false;
-}
+  NSURL* url = [NSURL URLWithString:urlString];
+  if (!url) {
+    return nil;
+  }
 
-- (bool)loadUrl:(NSString*)url {
-  return [self loadUrl:url withHeaders:[NSMutableDictionary dictionary]];
-}
+  NSMutableURLRequest* request = [NSMutableURLRequest requestWithURL:url];
+
+  NSString* httpMethod = requestParameters[@"method"];
+  if (httpMethod) {
+    [request setHTTPMethod:httpMethod];
+  }
 
-- (bool)loadUrl:(NSString*)url withHeaders:(NSDictionary<NSString*, NSString*>*)headers {
-  NSURL* nsUrl = [NSURL URLWithString:url];
-  if (!nsUrl) {
-    return false;
+  id httpBody = requestParameters[@"body"];
+  if ([httpBody isKindOfClass:[FlutterStandardTypedData class]]) {
+    [request setHTTPBody:[httpBody data]];
```

Comment:
What would happen if someone creates a GET request but still specifies a body? Will the `WKWebView` simply ignore the body and still load the page or does it result into an error / crash?

Reply:
This is currently handled through the NavigationDelegate's "onWebResourceError" callback. The error is passed through and can optionally be handled, but no hard crash.